src/builder/components/question/editable-title/survey-question-editable-title.component.tsx

```diff
@@ -0,0 +1,90 @@
+import { Button, Input } from 'antd';
+import React, { Component, RefObject } from 'react';
+
+export interface EditableTitleProps {
+    onTitleUpdated: (updatedTitle: string) => void;
+    content: string;
+    heading: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'h7';
+}
+
+export interface EditableTitleState {
+    editing: boolean;
+    value: string;
+    newValue: string;
+}
+
+export class EditableTitle extends Component<EditableTitleProps, EditableTitleState> {
+    state = {
+        editing: false,
+        value: this.props.content,
+        newValue: this.props.content,
+    };
+
+    constructor(props: EditableTitleProps) {
+        super(props);
+        this.inputRef = React.createRef();
```

Comment:
Why is this necessary?

Reply:
I was experimenting with refs. Still quite unsure where they would come in handy. I'll remove this because this seems useless

Reply:
Getting the position of elements would be an example - like automatically hiding/showing stuff when the element's `.top` reaches a certain value, or `scrollIntoView`